packages/flutter/lib/src/material/stepper.dart

```diff
@@ -577,11 +577,14 @@ class _StepperState extends State<Stepper> with TickerProviderStateMixin {
   }
 
   Widget _buildVertical() {
-    final List<Widget> children = <Widget>[];
 
-    for (int i = 0; i < widget.steps.length; i += 1) {
-      children.add(
-        Column(
+    return ListView.builder(
+      shrinkWrap: true,
+      itemCount: widget.steps.length,
+      itemBuilder: (BuildContext context, int i){
+
+        i < _keys.length ? _keys[i] : _keys.add(GlobalKey());
```

Comment:
Why are we introducing a GlobalKey here? This seems like it will introduce some pretty serious overhead...